packages/terra-notification-dialog/src/NotificationDialog.jsx

```diff
@@ -111,188 +78,139 @@ const propTypes = {
   if (!acceptAction && !rejectAction) {
     return null;
   }
+
+  const actionButtons = [];
+
   if (acceptAction) {
-    acceptButton = (emphasizedAction === 'accept')
-      ? <Button text={acceptAction.text} variant={ButtonVariants.EMPHASIS} onClick={acceptAction.onClick} />
-      : <Button text={acceptAction.text} onClick={acceptAction.onClick} />;
-  }
-  if (rejectAction) {
-    rejectButton = (emphasizedAction === 'reject')
-      ? <Button text={rejectAction.text} variant={ButtonVariants.EMPHASIS} onClick={rejectAction.onClick} />
-      : <Button text={rejectAction.text} onClick={rejectAction.onClick} />;
+    const buttonVariant = emphasizedAction === 'accept' ? { variant: 'emphasis' } : {};
+    actionButtons.push(<Button {...buttonVariant} data-terra-notification-dialog-button="accept" key="accept" text={acceptAction.text} onClick={acceptAction.onClick} />);
   }
 
-  if (buttonOrder === 'rejectFirst') {
-    return (
-      <div className={cx('actions')}>
-        {rejectButton}
-        {acceptButton}
-      </div>
-    );
+  if (rejectAction) {
+    const buttonVariant = emphasizedAction === 'reject' ? { variant: 'emphasis' } : {};
+    actionButtons.push(<Button {...buttonVariant} data-terra-notification-dialog-button="reject" key="reject" text={rejectAction.text} onClick={rejectAction.onClick} />);
   }
 
   return (
     <div className={cx('actions')}>
-      {acceptButton}
-      {rejectButton}
+      {buttonOrder === 'acceptFirst' ? actionButtons : actionButtons.reverse()}
     </div>
   );
 };
 
-const getIcon = (variant, customIcon = null) => {
-  switch (variant) {
-    case variants.ALERT:
-      return (<span className={cx(['icon', 'alert'])} />);
-    case variants.ERROR:
-      return (<span className={cx(['icon', 'error'])} />);
-    case variants.WARNING:
-      return (<span className={cx(['icon', 'warning'])} />);
-    case variants.INFO:
-      return (<span className={cx(['icon', 'info'])} />);
-    case variants.SUCCESS:
-      return (<span className={cx(['icon', 'success'])} />);
-    case variants.CUSTOM:
-      return customIcon;
-    default:
-      return null;
-  }
-};
-
-class NotificationDialog extends React.Component {
-  constructor(props) {
-    super(props);
-
-    this.handleKeydown = this.handleKeydown.bind(this);
-    this.escapeKey = KeyCode.KEY_ESCAPE;
-  }
-
-  componentDidMount() {
-    document.addEventListener('keydown', this.handleKeydown);
-  }
-
-  componentWillUnmount() {
-    document.removeEventListener('keydown', this.handleKeydown);
+const NotificationDialog = (props) => {
+  const theme = React.useContext(ThemeContext);
+
+  const {
+    dialogTitle,
+    startMessage,
+    endMessage,
+    content,
+    acceptAction,
+    rejectAction,
+    variant,
+    isOpen,
+    buttonOrder,
+    emphasizedAction,
+    custom,
+    ...customProps
+  } = props;
+
+  if (process.env.NODE_ENV !== 'production' && acceptAction === undefined && rejectAction === undefined) {
```

Comment:
If we somehow get to prod with this error in place, I think we should still throw the error and let the error boundary take over. The user would be stuck.

Reply:
agree. I made this change in terra-app. Will do it here as well.

Reply:
Same for the next one too IMO

Reply:
I see what I did. I just missed update this condition. 🤦‍♀️  Fixed: https://github.com/cerner/terra-framework/pull/1105/commits/c45612d9b69bb2982f5ca4adbcad9abaf6a161f5#diff-cb62dc7ce0197bc569a9952450bb9fe5L149